integration-tests/tests/utils/clp_mode_utils.py

```diff
@@ -0,0 +1,41 @@
+"""Provides utilities related to the user-level configurations of CLP's operating modes."""
+
+from collections.abc import Callable
+
+from clp_py_utils.clp_config import (
+    ClpConfig,
+    Package,
+    QueryEngine,
+    StorageEngine,
+)
+
+CLP_MODE_CONFIGS: dict[str, Callable[[], ClpConfig]] = {
+    "clp-text": lambda: ClpConfig(
```

Comment:
mind explaining why we need to return a callable rather than an actual ClpConfig object?